packages/core/factories/src/useCategoryFactory.ts

```diff
@@ -10,16 +10,16 @@ export function useCategoryFactory<CATEGORY, CATEGORY_SEARCH_PARAMS>(
   factoryParams: UseCategoryFactoryParams<CATEGORY, CATEGORY_SEARCH_PARAMS>
 ) {
   return function useCategory(cacheId?: string): UseCategory<CATEGORY> {
-    const { state, persistedResource } = usePersistedState(cacheId);
+    const { state } = useSSR(cacheId);
     const categories: Ref<CATEGORY[]> = ref(state || []);
     const loading = ref(false);
 
     const search = async (params: CATEGORY_SEARCH_PARAMS) => {
-      loading.value = true;
-      categories.value = await persistedResource<CATEGORY[]>(
-        factoryParams.categorySearch,
-        params
-      );
+      if (!state) {
+        loading.value = true;
+      }
+      categories.value = await factoryParams.categorySearch(params);
+      eventBus.emit('set-ssr-cache', { key: cacheId, value: categories.value });
```

Comment:
after other comment
```suggestion
      setCache(categories.value)
```
wdyt @andrzejewsky @filrak ?